flake8_async.py

```diff
@@ -6,11 +6,22 @@
 __version__ = "22.3.10"
 
 ASYNC100 = "ASYNC100: sync HTTP call in async function should use httpx.AsyncClient"
+ASYNC101 = "ASYNC101: blocking sync call in async function"
 
 
 class Visitor(ast.NodeVisitor):
-    PACKAGES = ("requests", "httpx")
-    METHODS = tuple("get options head post put patch delete request send".split())
+    HTTP_PACKAGES = ("requests", "httpx")
+    HTTP_METHODS = tuple("get options head post put patch delete request send".split())
+    SUBPROCESS_METHODS = (
+        "run",
+        "Popen",
+        # deprecated methods
+        "call",
+        "check_call",
+        "check_output",
+        "getoutput",
+        "getstatusoutput",
+    )
```

Comment:
Reviewing this, I realize that I really want to provide suggested alternatives - for many users, being told "you shouldn't use `subprocess.run()` in an async function" will immediately make them ask "well, what *should* I use then?"

...and that gets tricky pretty quickly.  The suggested alternative will be different depending on your async library, and as a static tool we can't just use `sniffio`.  On the other hand we *can* detect `import {anyio, trio, asyncio}` statements and make some assumptions (plus maybe a config option) like using the preference order above if you import more than one.

OK, so for `time.sleep()` we recommend `await lib.sleep()` and life is good.  But subprocess handling is much *more* varied between frameworks, and that's without even considering file IO or async path handling!

------------

So I'm wondering if we should instead just lean into the tension, merge this, declare that `flake8-async` is for `asyncio` - making #5 easier too! - and we can add similar logic with specific suggested alternatives to `flake8-trio` later.

@cooperlees sorry to ping you when your low on time, but thoughts?

Reply:
Well, we could refer them to the repo and write a longer description in the readme (which would mostly boil down to "read the documentation of your async library"). Detecting the library sounds a bit tenuous.
So something like
> blocking subprocess call in async function. Consider using `asyncio.create_subprocess_exec()` or similar, check https://github.com/cooperlees/flake8-async for more information.

Could add additional logic to suggest `create_subprocess_shell` if `Popen` with `shell=True`.

Reply:
Pulled out the discussion on what direction to take into #7 - felt a bit silly to post a wall of text hidden in this PR

Reply:
I vote just state use the subprocess alternative for your async framework. I envision this to be the generic async linter ...